cypress/e2e/settings/labs-user-settings-tab.spec.ts

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 Suguru Hirahara
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { HomeserverInstance } from "../../plugins/utils/homeserver";
+
+describe("Labs user settings tab", () => {
+    let homeserver: HomeserverInstance;
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, "Hanako");
+        });
+
+        // Show labs settings
+        cy.tweakConfig({ show_labs_settings: "true" });
+
+        cy.openUserSettings("Labs");
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should be rendered properly", () => {
+        cy.get(".mx_SettingsTab.mx_LabsUserSettingsTab").percySnapshotElement("User settings tab - Labs", {
```

Comment:
I'm not sure this is a good candidate for a percy snapshot, or an e2e test at all, as the labs screen is intended to change regularly

Reply:
Regularly but not daily, right? I believe that taking a whole snapshot would help to prevent a case like https://github.com/vector-im/element-web/issues/24813. Please Let me know if a Jest snapshot would be sufficient here.

Reply:
@andybalaam I would like to ask your opinion on that too. Thanks in advance.

Reply:
In my opinion a jest test is more suitable for this case - where we can use stable mock data that covers all the cases. Then we can be sure that the case from 24813 will be covered and continue to be covered, independent of what features are in labs at that time.

Reply:
Yeah, I agree that a Jest test would probably be the best fit here. We have to review each Percy failure manually, so we should definitely not create tests that we expect to fail regularly.

Thanks for the work @luixxiul !

Reply:
Thanks both of you for opinions! Now I am convinced.